src/test/javascript/spec/component/course/course-exercises.component.spec.ts

```diff
@@ -0,0 +1,181 @@
+                { provide: TranslateService, useClass: MockTranslateService },
+                {
+                    provide: ActivatedRoute,
+                    useValue: route,
+                },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(CourseExercisesComponent);
+                component = fixture.componentInstance;
+                service = TestBed.inject(CourseManagementService);
+                courseCalculation = TestBed.inject(CourseScoreCalculationService);
+                translateService = TestBed.inject(TranslateService);
+                exerciseService = TestBed.inject(ExerciseService);
+                localStorageService = TestBed.inject(LocalStorageService);
+
+                course = new Course();
+                course.id = 123;
+                exercise = new ModelingExercise(UMLDiagramType.ClassDiagram, course, undefined) as Exercise;
+                exercise.dueDate = moment('2021-01-13T16:11:00+01:00').add(1, 'days');
+                exercise.releaseDate = moment('2021-01-13T16:11:00+01:00').subtract(1, 'days');
+                course.exercises = [exercise];
+                spyOn(service, 'getCourseUpdates').and.returnValue(of(course));
+                spyOn(translateService, 'onLangChange').and.returnValue(of(new Subject()));
+                spyOn(localStorageService, 'retrieve').and.returnValue('OVERDUE,NEEDS_WORK');
+                courseCalculationSpy = stub(courseCalculation, 'getCourse').returns(course);
+
+                fixture.detectChanges();
+            });
+    });
+
+    afterEach(() => {
+        component.ngOnDestroy();
+        sinon.restore();
+    });
+
+    it('should initialize', () => {
+        expect(component.course).to.deep.equal(course);
+        expect(courseCalculationSpy.callCount).to.equal(2);
+        expect(courseCalculationSpy.getCall(0).calledWithExactly(course.id)).to.be.true;
+        expect(courseCalculationSpy.getCall(1).calledWithExactly(course.id)).to.be.true;
+    });
+
+    it('should invoke setSortingAttribute', () => {
+        const sortingButton = fixture.debugElement.query(By.css('#dueDateSorting'));
+        expect(sortingButton).to.exist;
+        sortingButton.nativeElement.click();
+        expect(component.sortingAttribute).to.equal(SortingAttribute.DUE_DATE);
+    });
+
+    it('should react to changes', () => {
+        spyOn(exerciseService, 'getNextExerciseForHours').and.returnValue(exercise);
+        component.ngOnChanges();
+        expect(component.nextRelevantExercise).to.deep.equal(exercise);
+    });
+
+    it('should reorder all exercises', () => {
+        component.sortingOrder = ExerciseSortingOrder.DESC;
+        const sortingButton = fixture.debugElement.query(By.css('#flip'));
+        expect(sortingButton).to.exist;
+        sortingButton.nativeElement.click();
+        expect(component.sortingOrder).to.equal(ExerciseSortingOrder.ASC);
```

Comment:
You are not really asserting that the sorting worked here. I would expect to see some test that asserts that the sorting really happened

Reply:
Extended this test with some further expectations and calls, hope this hits the point what you expected (80405fbc1dea9ebf80621ffc70f83f0569ecd7bb)